include/seqan3/core/concept/tuple.hpp

```diff
@@ -49,14 +49,19 @@ SEQAN3_CONCEPT tuple_get = requires (tuple_t & v, tuple_t const & v_c)
     requires std::tuple_size_v<tuple_t> > 0;
 
     typename std::tuple_element<0, tuple_t>::type;
-    { get<0>(v)              } -> typename std::tuple_element<0, tuple_t>::type;
+
+    SEQAN3_RETURN_TYPE_CONTRAINT(get<0>(v), std::convertible_to, typename std::tuple_element<0, tuple_t>::type);
 //     requires weakly_assignable_from<decltype(get<0>(v)), typename std::tuple_element<0, tuple_t>::type>;
     //TODO check that the previous returns something that can be assigned to
-    // unfortunately std::assignable_from requires lvalue-reference, but we want to accept xvalues too (returned proxies)
-    { get<0>(v_c)            } -> typename std::tuple_element<0, tuple_t>::type;
-    { get<0>(std::move(v))   } -> typename std::tuple_element<0, tuple_t>::type;
+    // unfortunately std::assignable_from requires lvalue-reference, but we want to accept xvalues too (returned
+    // proxies)
+    SEQAN3_RETURN_TYPE_CONTRAINT(get<0>(v_c), std::convertible_to, typename std::tuple_element<0, tuple_t>::type);
+    SEQAN3_RETURN_TYPE_CONTRAINT(get<0>(std::move(v)),
+                                 std::convertible_to, typename std::tuple_element<0, tuple_t>::type);
     // TODO: The return type for std::tuple is wrong until gcc-8.0, for gcc > 8.0 this is fixed.
     { get<0>(std::move(v_c)) };// -> typename std::tuple_element<0, tuple_t>::type const &&;
+    // SEQAN3_RETURN_TYPE_CONTRAINT(get<0>(std::move(v_c)),
+    //                              std::convertible_to, typename std::tuple_element<0, tuple_t>::type const &&);
```

Comment:
:question: before it was

```c++
{ get<0>(std::move(v_c)) };// -> typename std::tuple_element<0, tuple_t>::type const &&;
```

Now I added a version with `SEQAN3_RETURN_TYPE_CONTRAINT`, which is still commented out, because we would need to wrap that in `if gcc > 8.0` (assuming the comment is correct; there is no reference to a gcc ticket.).

I can remove that again if you want.